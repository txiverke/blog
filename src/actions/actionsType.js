// @flow

export const LOAD_STATS_REQUEST = 'LOAD_STATS_REQUEST'
export const LOAD_STATS_SUCCESS = 'LOAD_STATS_SUCCESS'
export const LOAD_STATS_FAILURE = 'LOAD_STATS_FAILURE'

export const LOG_IN_USER_REQUEST = 'LOG_IN_USER_REQUEST'
export const LOG_IN_USER_SUCCESS = 'LOG_IN_USER_SUCCESS'
export const LOG_IN_USER_FAILURE = 'LOG_IN_USER_FAILURE'

export const IS_AUTHENTICATED_REQUEST = 'IS_AUTHENTICATE_REQUEST'
export const IS_AUTHENTICATED_SUCCESS = 'IS_AUTHENTICATE_SUCCESS'
export const IS_AUTHENTICATED_FAILURE = 'IS_AUTHENTICATE_FAILURE'

export const LOAD_USER_DATA_REQUEST = 'LOAD_USER_DATA_REQUEST'
export const LOAD_USER_DATA_SUCCESS = 'LOAD_USER_DATA_SUCCESS'
export const LOAD_USER_DATA_FAILURE = 'LOAD_USER_DATA_FAILURE'

export const UPDATE_USER_DATA_REQUEST = 'UPDATE_USER_DATA_REQUEST'
export const UPDATE_USER_DATA_SUCCESS = 'UPDATE_USER_DATA_SUCCESS'
export const UPDATE_USER_DATA_FAILURE = 'UPDATE_USER_DATA_FAILURE'

export const LOG_OUT_USER_REQUEST = 'LOG_OUT_USER_REQUEST'
export const LOG_OUT_USER_SUCCESS = 'LOG_OUT_USER_SUCCESS'
export const LOG_OUT_USER_FAILURE = 'LOG_OUT_USER_FAILURE'

export const LOAD_POST_DATA_REQUEST = 'LOAD_POST_DATA_REQUEST'
export const LOAD_POST_DATA_SUCCESS = 'LOAD_POST_DATA_SUCCESS'
export const LOAD_POST_DATA_FAILURE = 'LOAD_POST_DATA_FAILURE'

export const LOAD_POST_ITEM_REQUEST = 'LOAD_POST_ITEM_REQUEST'
export const LOAD_POST_ITEM_SUCCESS = 'LOAD_POST_ITEM_SUCCESS'
export const LOAD_POST_ITEM_FAILURE = 'LOAD_POST_ITEM_FAILURE'
export const UPDATE_POST_ITEM_SUCCESS = 'UPDATE_POST_ITEM_SUCCESS'

export const CREATE_POST_DATA_REQUEST = 'CREATE_POST_DATA_REQUEST'
export const CREATE_POST_DATA_SUCCESS = 'CREATE_POST_DATA_SUCCESS'
export const CREATE_POST_DATA_FAILURE = 'CREATE_POST_DATA_FAILURE'

export const UPDATE_POST_DATA_REQUEST = 'UPDATE_POST_DATA_REQUEST'
export const UPDATE_POST_DATA_SUCCESS = 'UPDATE_POST_DATA_SUCCESS'
export const UPDATE_POST_DATA_FAILURE = 'UPDATE_POST_DATA_FAILURE'

export const REMOVE_POST_DATA_REQUEST = 'REMOVE_POST_DATA_REQUEST'
export const REMOVE_POST_DATA_SUCCESS = 'REMOVE_POST_DATA_SUCCESS'
export const REMOVE_POST_DATA_FAILURE = 'REMOVE_POST_DATA_FAILURE'
