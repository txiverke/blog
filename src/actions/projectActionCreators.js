// @flow

import * as ACTION from './actionsType'
import { setPromise } from '../utils/helpers'

const URL = 'projects'

/**
 * LOAD PROJECTS
 */

 export const loadProjectDataRequest = () => ({ type: ACTION.LOAD_PROJECT_DATA_REQUEST })
 export const loadProjectDataSuccess = (payload: Array<Object>) => ({ type: ACTION.LOAD_PROJECT_DATA_SUCCESS, payload })
 export const loadProjectDataFailure = () => ({ type: ACTION.LOAD_PROJECT_DATA_FAILURE })

 export const loadProjectData = () => async (dispatch: Function) => {
  dispatch(loadProjectDataRequest())

  try {
    setPromise.method = 'GET'
    setPromise.body = null
    setPromise.urls = URL
    const data = await setPromise.response()
    return dispatch(loadProjectDataSuccess(data))
  } catch (err) {
    dispatch(loadProjectDataFailure())
  }
 }