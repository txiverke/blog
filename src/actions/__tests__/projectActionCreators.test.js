import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import * as action from '../projectActionCreators'
import * as type from '../actionsType'

const middlewares = [ thunk ]
const mockStore = configureMockStore(middlewares)

describe('projectActionCreators', () => {
  describe('load project actions', () => {
    it('should return LOAD_PROJECT_DATA_REQUEST action', () => {
      expect(action.loadProjectDataRequest()).toMatchSnapshot()
    })

    it('should return LOAD_PROJECT_DATA_SUCCESS action', () => {
      expect(action.loadProjectDataSuccess(['test'])).toMatchSnapshot()
    })

    it('should return LOAD_PROJECT_DATA_FAILURE action', () => {
      expect(action.loadProjectDataFailure()).toMatchSnapshot()
    })

    it('expected actions should be dispatched on request', () => {
      const store = mockStore({})
      const expectedActions = [
        'LOAD_PROJECT_DATA_REQUEST',
        'LOAD_PROJECT_DATA_FAILURE'
      ]

      return store.dispatch(action.loadProjectData())
        .then(() => {
          const actualActions = store.getActions().map(action => action.type)
          expect(actualActions).toEqual(expectedActions)
       })
    })
  })

  describe('load project item actions', () => {
    it('should return LOAD_PROJECT_ITEM_REQUEST action', () => {
      expect(action.loadProjectItemRequest()).toMatchSnapshot()
    })

    it('should return LOAD_PROJECT_ITEM_SUCCESS action', () => {
      expect(action.loadProjectItemSuccess({test: true})).toMatchSnapshot()
    })

    it('should return LOAD_PROJECT_ITEM_FAILURE action', () => {
      expect(action.loadProjectItemFailure()).toMatchSnapshot()
    })

    it('expected actions should be dispatched on request', () => {
      const store = mockStore({})
      const expectedActions = [
        'LOAD_PROJECT_ITEM_REQUEST',
        'LOAD_PROJECT_ITEM_FAILURE'
      ]

      return store.dispatch(action.loadProjectItem('test'))
        .then(() => {
          const actualActions = store.getActions().map(action => action.type)
          expect(actualActions).toEqual(expectedActions)
       })
    })
  })

  describe('create project actions', () => {
    it('should return CREATE_PROJECT_DATA_REQUEST action', () => {
      expect(action.createProjectDataRequest()).toMatchSnapshot()
    })

    it('should return CREATE_PROJECT_DATA_SUCCESS action', () => {
      expect(action.createProjectDataSuccess(['test'])).toMatchSnapshot()
    })

    it('should return CREATE_PROJECT_DATA_FAILURE action', () => {
      expect(action.createProjectDataFailure()).toMatchSnapshot()
    })

    it('expected actions should be dispatched on request', () => {
      const store = mockStore({})
      const expectedActions = [
        'CREATE_PROJECT_DATA_REQUEST',
        'CREATE_PROJECT_DATA_FAILURE'
      ]

      return store.dispatch(action.createProjectData({test: true}))
        .then(() => {
          const actualActions = store.getActions().map(action => action.type)
          expect(actualActions).toEqual(expectedActions)
       })
    })
  })

  describe('update project actions', () => {
    it('should return UPDATE_PROJECT_DATA_REQUEST action', () => {
      expect(action.updateProjectDataRequest()).toMatchSnapshot()
    })

    it('should return UPDATE_PROJECT_DATA_SUCCESS action', () => {
      expect(action.updateProjectDataSuccess(['test'])).toMatchSnapshot()
    })

    it('should return UPDATE_PROJECT_DATA_FAILURE action', () => {
      expect(action.updateProjectDataFailure()).toMatchSnapshot()
    })

    it('expected actions should be dispatched on request', () => {
      const store = mockStore({})
      const expectedActions = [
        'UPDATE_PROJECT_DATA_REQUEST',
        'UPDATE_PROJECT_DATA_FAILURE'
      ]

      return store.dispatch(action.updateProjectData({test: true}, 'id'))
        .then(() => {
          const actualActions = store.getActions().map(action => action.type)
          expect(actualActions).toEqual(expectedActions)
       })
    })
  })

  describe('remove project actions', () => {
    it('should return REMOVE_PROJECT_DATA_REQUEST action', () => {
      expect(action.removeProjectDataRequest()).toMatchSnapshot()
    })

    it('should return REMOVE_PROJECT_DATA_SUCCESS action', () => {
      expect(action.removeProjectDataSuccess('id')).toMatchSnapshot()
    })

    it('should return REMOVE_PROJECT_DATA_FAILURE action', () => {
      expect(action.removeProjectDataFailure()).toMatchSnapshot()
    })

    it('expected actions should be dispatched on request', () => {
      const store = mockStore({})
      const expectedActions = [
        'REMOVE_PROJECT_DATA_REQUEST',
        'REMOVE_PROJECT_DATA_FAILURE'
      ]

      return store.dispatch(action.removeProjectData('id'))
        .then(() => {
          const actualActions = store.getActions().map(action => action.type)
          expect(actualActions).toEqual(expectedActions)
       })
    })
  })
})