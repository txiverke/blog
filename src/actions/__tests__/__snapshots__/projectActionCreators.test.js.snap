// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`projectActionCreators create project actions should return CREATE_PROJECT_DATA_FAILURE action 1`] = `
Object {
  "type": "CREATE_PROJECT_DATA_FAILURE",
}
`;

exports[`projectActionCreators create project actions should return CREATE_PROJECT_DATA_REQUEST action 1`] = `
Object {
  "type": "CREATE_PROJECT_DATA_REQUEST",
}
`;

exports[`projectActionCreators create project actions should return CREATE_PROJECT_DATA_SUCCESS action 1`] = `
Object {
  "payload": Array [
    "test",
  ],
  "type": "CREATE_PROJECT_DATA_SUCCESS",
}
`;

exports[`projectActionCreators load project actions should return LOAD_PROJECT_DATA_FAILURE action 1`] = `
Object {
  "type": "LOAD_PROJECT_DATA_FAILURE",
}
`;

exports[`projectActionCreators load project actions should return LOAD_PROJECT_DATA_REQUEST action 1`] = `
Object {
  "type": "LOAD_PROJECT_DATA_REQUEST",
}
`;

exports[`projectActionCreators load project actions should return LOAD_PROJECT_DATA_SUCCESS action 1`] = `
Object {
  "payload": Array [
    "test",
  ],
  "type": "LOAD_PROJECT_DATA_SUCCESS",
}
`;

exports[`projectActionCreators load project item actions should return LOAD_PROJECT_ITEM_FAILURE action 1`] = `
Object {
  "type": "LOAD_PROJECT_ITEM_FAILURE",
}
`;

exports[`projectActionCreators load project item actions should return LOAD_PROJECT_ITEM_REQUEST action 1`] = `
Object {
  "type": "LOAD_PROJECT_ITEM_REQUEST",
}
`;

exports[`projectActionCreators load project item actions should return LOAD_PROJECT_ITEM_SUCCESS action 1`] = `
Object {
  "payload": Object {
    "test": true,
  },
  "type": "LOAD_PROJECT_ITEM_SUCCESS",
}
`;

exports[`projectActionCreators remove project actions should return REMOVE_PROJECT_DATA_FAILURE action 1`] = `
Object {
  "type": "REMOVE_PROJECT_DATA_FAILURE",
}
`;

exports[`projectActionCreators remove project actions should return REMOVE_PROJECT_DATA_REQUEST action 1`] = `
Object {
  "type": "REMOVE_PROJECT_DATA_REQUEST",
}
`;

exports[`projectActionCreators remove project actions should return REMOVE_PROJECT_DATA_SUCCESS action 1`] = `
Object {
  "payload": "id",
  "type": "REMOVE_PROJECT_DATA_SUCCESS",
}
`;

exports[`projectActionCreators update project actions should return UPDATE_PROJECT_DATA_FAILURE action 1`] = `
Object {
  "type": "UPDATE_PROJECT_DATA_FAILURE",
}
`;

exports[`projectActionCreators update project actions should return UPDATE_PROJECT_DATA_REQUEST action 1`] = `
Object {
  "type": "UPDATE_PROJECT_DATA_REQUEST",
}
`;

exports[`projectActionCreators update project actions should return UPDATE_PROJECT_DATA_SUCCESS action 1`] = `
Object {
  "payload": Array [
    "test",
  ],
  "type": "UPDATE_PROJECT_DATA_SUCCESS",
}
`;
